pkg/iris/events.go

```diff
@@ -25,78 +25,86 @@ func GenerateRunId() string {
 }
 
 type EventBuilder interface {
-	Cluster(cluster models.Cluster) EventBuilder
-	Id(id string) EventBuilder
-	Scope(scope string) EventBuilder
-	Type(eventType string) EventBuilder
-	Source(source string) EventBuilder
-	Time(time string) EventBuilder
-	Subject(subject string) EventBuilder
+	Header(header models.HeaderProperties) EventBuilder
+	Body(body models.DiscoveryItem) EventBuilder
 
-	Build() models.DiscoveryItem
+	Build() models.DiscoveryEvent
+}
+
+type CommandBuilder interface {
+	Header(header models.CommandProperties) CommandBuilder
+	Body(body models.CommandBody) CommandBuilder
+
+	Build() models.CommandEvent
 }
 
 type eventBuilder struct {
-	c         models.Cluster
-	id        string
-	scope     string
-	eventType string
-	source    string
-	time      string
-	subject   string
+	header models.HeaderProperties
+	body   models.DiscoveryItem
 }
 
-func New() EventBuilder {
-	return &eventBuilder{}
+type commandBuilder struct {
+	header models.CommandProperties
+	body   models.CommandBody
 }
 
-func (eb *eventBuilder) Cluster(cluster models.Cluster) EventBuilder {
-	eb.c = cluster
-	return eb
+// Discovery Event Builder
+func New() EventBuilder {
+	return &eventBuilder{}
 }
 
-func (eb *eventBuilder) Id(id string) EventBuilder {
-	eb.id = id
+func (eb *eventBuilder) Header(header models.HeaderProperties) EventBuilder {
+	eb.header = header
 	return eb
 }
 
-func (eb *eventBuilder) Scope(scope string) EventBuilder {
-	eb.scope = scope
+func (eb *eventBuilder) Body(body models.DiscoveryItem) EventBuilder {
+	eb.body = body
 	return eb
 }
 
-func (eb *eventBuilder) Type(eventType string) EventBuilder {
-	eb.eventType = eventType
-	return eb
+func (eb *eventBuilder) Build() models.DiscoveryEvent {
+	body := &models.DiscoveryItem{
+		State: eb.body.State,
+	}
+	header := &models.HeaderProperties{
+		HeaderClass: eb.header.HeaderClass,
+		HeaderType:  eb.header.HeaderType,
+		HeaderScope: eb.header.HeaderScope,
+		HeaderId:    eb.header.HeaderId,
```

Comment:
Do we need the Header* prefix on all attributes in the HeaderProperties?

Reply:
Ahh, not anymore! I added the header prefix to easily distinguish between old vs new back when the old code was still there. I'll clean that up 👍